### PR TITLE
Refactoring build process

### PR DESCRIPTION
**Pull request by maia-iyer**
Deprecating images that we will no longer support and cleaning Makefile

### CODE REVIEW

File: .github/workflows/master-build.yaml [hunk 1]
```diff
@@ -46,17 +43,12 @@ jobs:
...
-      # Create tagged versioned images
-      # - name: Push artifacts for tornjak-backend-spire-multiversions
-      #   run: make release-tornjak-backend-spire-multiversions-ghcr
       - name: Build and push tornjak backend image
-        run: make release-tornjak-backend-ghcr
+        run: make release-tornjak-backend
       - name: Build and push tornjak frontend image
-        run: make release-tornjak-frontend-ghcr
+        run: make release-tornjak-frontend
       - name: Build and push tornjak image
```
Comment (mrsabath) [inline .github/workflows/master-build.yaml:50]:
```suggestion
      - name: Build and push tornjak image (frontend+backend)
```

File: Makefile [hunk 2]
```diff
@@ -51,24 +50,21 @@ vendor:
 
 
 # Containerized components
+## Backend container
```
Comment (mrsabath) [inline Makefile:53]:
```suggestion
## Build Backend container
```

File: Makefile [hunk 3]
```diff
@@ -51,24 +50,21 @@ vendor:
...
 container-tornjak-backend: bin/tornjak-backend
 	docker build --no-cache -f Dockerfile.backend-container --build-arg version=$(VERSION) -t ${CONTAINER_BACKEND_TAG} .
```
Comment (mrsabath) [inline Makefile:56]:
```suggestion

## Build and push Backend to image repository
```

File: Makefile [hunk 4]
```diff
@@ -51,24 +50,21 @@ vendor:
...
 container-tornjak-backend-push: container-tornjak-backend
 	docker push ${CONTAINER_BACKEND_TAG}
 
-container-tornjak-backend-spire: bin/tornjak-backend
-	docker build --no-cache -f Dockerfile.add-backend --build-arg version=$(VERSION) -t ${CONTAINER_BACKEND_WITH_SPIRE_TAG} .
-
-container-tornjak-backend-spire-push: container-tornjak-backend-spire
-	docker push ${CONTAINER_BACKEND_WITH_SPIRE_TAG}
-
+## Manager container
```
Comment (mrsabath) [inline Makefile:60]:
```suggestion
## Build Manager container
```

File: Makefile [hunk 5]
```diff
@@ -51,24 +50,21 @@ vendor:
...
 container-manager: bin/tornjak-manager #ui-manager
 	docker build --no-cache -f Dockerfile.tornjak-manager --build-arg version=$(VERSION) -t ${CONTAINER_MANAGER_TAG} .
```
Comment (mrsabath) [inline Makefile:63]:
```suggestion
## Build and push Manager to image repository
```

File: Makefile [hunk 6]
```diff
@@ -51,24 +50,21 @@ vendor:
...
 
 container-manager-push: container-manager
 	 docker push ${CONTAINER_MANAGER_TAG}
 
+## Frontend container
```
Comment (mrsabath) [inline Makefile:67]:
```suggestion
## Build Frontend container
```

File: Makefile [hunk 7]
```diff
@@ -51,24 +50,21 @@ vendor:
...
 container-frontend: #ui-agent 
 	docker build --no-cache -f Dockerfile.frontend-container --build-arg version=$(VERSION) -t ${CONTAINER_FRONTEND_TAG} .
```
Comment (mrsabath) [inline Makefile:70]:
```suggestion

## Build and push Frontend to image repository
```

File: Makefile [hunk 8]
```diff
@@ -79,53 +75,50 @@ compose-frontend:
 container-frontend-push: container-frontend
 	docker push ${CONTAINER_FRONTEND_TAG}
 
-# WARNING: EXPERIMENTAL feature to merge frontend and backend in one container
+## Backend + Frontend container
```
Comment (mrsabath) [inline Makefile:78]:
```suggestion
## Build tornjak container (Backend + Frontend)
```

File: Makefile [hunk 9]
```diff
@@ -79,53 +75,50 @@ compose-frontend:
...
 container-tornjak: bin/tornjak-backend #ui-agent
 	docker build --no-cache -f Dockerfile.tornjak-container --build-arg version=$(VERSION) -t ${CONTAINER_TAG} .
```
Comment (mrsabath) [inline Makefile:81]:
```suggestion

## Build and push tornjak (Backend + Frontend) to image repository
```

File: Makefile [hunk 10]
```diff
@@ -1,20 +1,19 @@
-.PHONY: ui vendor build container-tornjak-backend-spire container-tornjak-backend-spire-push container-manager container-manager-push release-tornjak-backend-spire-multiversions push container-frontend container-frontend-push container-tornjak-backend container-tornjak-backend-push
+.PHONY: ui vendor build container-tornjak-backend-spire container-manager container-manager-push push container-frontend container-frontend-push container-tornjak-backend container-tornjak-backend-push
 
 VERSION=$(shell cat version.txt)
 
+## when containers are built, they are tagged with these
 CONTAINER_TAG ?= tsidentity/tornjak:$(VERSION)
```
Comment (mrsabath) [inline Makefile:6]: I think the confusion starts here, with the variable names. 
Perhaps we should have the following naming convention:
```
CONTAINER_BACKEND_IMAGEPATH_TEST ?= tsidentity/tornjak-backend
CONTAINER_BACKEND_IMAGEPATH ?= ghcr.io/spiffe/tornjak-backend
```

And then we use (and tag) them accordingly?

File: Makefile [hunk 11]
```diff
@@ -79,53 +75,50 @@ compose-frontend:
...
 
-# releases for Github Container Registry
-release-tornjak-backend-ghcr: container-tornjak-backend
-	docker tag ${CONTAINER_BACKEND_TAG} ${CONTAINER_BACKEND_GHCR_IMAGEPATH}:latest
-	docker tag ${CONTAINER_BACKEND_TAG} ${CONTAINER_BACKEND_GHCR_IMAGEPATH}:$(VERSION)
+## BEGIN RELEASES FOR GITHUB CONTAINER REGISTRY ##
+# These targets are used by Github to create official pre-built images
+
+## backend image
+release-tornjak-backend: container-tornjak-backend
```
Comment (mrsabath) [inline Makefile:90]: So here, maybe we should rename it to
`container-backend-push-test` ? And then we tag and push accordingly?

File: Makefile [hunk 12]
```diff
@@ -79,53 +75,50 @@ compose-frontend:
...
-	done
-
-release-tornjak-frontend-ghcr: container-frontend
-	docker tag ${CONTAINER_FRONTEND_TAG} ${CONTAINER_FRONTEND_GHCR_IMAGEPATH}:latest
-	docker tag ${CONTAINER_FRONTEND_TAG} ${CONTAINER_FRONTEND_GHCR_IMAGEPATH}:$(VERSION)
+	docker push ${CONTAINER_BACKEND_IMAGEPATH}:latest
+	docker push ${CONTAINER_BACKEND_IMAGEPATH}:${VERSION}
+
+## frontend image
+release-tornjak-frontend: container-frontend
```
Comment (mrsabath) [inline Makefile:98]: same as above

**Review by mrsabath — CHANGES_REQUESTED** [review]

File: Makefile (hunk 11 above)
Comment (maia-iyer) [reply Makefile:93]: sticking with the `release` prefix because it corresponds to Github releases

File: .github/workflows/master-build.yaml [hunk 13]
```diff
@@ -31,34 +31,26 @@ jobs:
...
-      # Deprecating the backend+spire image for now
-      # - name: Set backend-spire image name
-      #   run: echo "CONTAINER_BACKEND_WITH_SPIRE_TAG=ghcr.io/${{ github.repository_owner }}/tornjak-backend-spire-server:$GITHUB_SHA" >> $GITHUB_ENV
+     
+      # set DEV tags to have GITHUB_SHA  
       - name: Set github commit id
         run: echo "GITHUB_SHA=$GITHUB_SHA" >> $GITHUB_ENV
       - name: Set backend image name
-        run: echo "CONTAINER_BACKEND_TAG=ghcr.io/${{ github.repository_owner }}/tornjak-backend:$GITHUB_SHA" >> $GITHUB_ENV
+        run: echo "CONTAINER_BACKEND_DEV_TAG=ghcr.io/${{ github.repository_owner }}/tornjak-backend:$GITHUB_SHA" >> $GITHUB_ENV
```
Comment (mrsabath) [inline .github/workflows/master-build.yaml:39]: shouldn't they be `*_RELEASE_TAG`?

Comment (maia-iyer) [reply .github/workflows/master-build.yaml:39]: no, the release tags are for the version, the dev tag is for the commit. 
the release tag is already set in the makefile
i see how this is confusing tho - we still have to push two image tags: one for with the version (currnetly RELEASE tag) and one with the git sha (currently the DEV tag as set here)

what would you propose we tag the latter if not "DEV"?

File: Makefile [hunk 14]
```diff
@@ -1,24 +1,25 @@
...
+## `make release-*` pushes to DEV tag as well as below corresponding RELEASE tag
+## used by Github
 
-CONTAINER_MANAGER_TAG ?= tsidentity/tornjak-manager:$(VERSION)
 GO_FILES := $(shell find . -type f -name '*.go' -not -name '*_test.go' -not -path './vendor/*')
 
-all: bin/tornjak-backend bin/tornjak-manager ui-manager container-manager container-frontend container-tornjak-backend
+all: bin/tornjak-backend bin/tornjak-manager container-manager container-frontend container-backend container-tornjak
+
+#### BEGIN LOCAL EXECUTABLE BUILDS ####
```
Comment (mrsabath) [inline Makefile:23]:
```suggestion

# build binaries for Tornjak backend (bin/tornjak-backend)
```

File: Makefile [hunk 15]
```diff
@@ -30,117 +31,73 @@ bin/tornjak-manager: $(GO_FILES) vendor
 	docker run --rm -v "${PWD}":/usr/src/myapp -w /usr/src/myapp -e GOOS=linux -e GOARCH=amd64 golang:1.16 /bin/sh -c "go build --tags 'sqlite_json' -o tornjak-manager tornjak-backend/cmd/manager/manager.go; go build --tags 'sqlite_json' -mod=vendor -ldflags '-s -w -linkmode external -extldflags "-static"' -o bin/tornjak-manager tornjak-backend/cmd/manager/manager.go"
```
Comment (mrsabath) [inline Makefile:33]:
```suggestion
# build Tornjak frontend binaries
```

**Review by mrsabath — APPROVED** [review]
This makes the build process much cleaner. Thank you @maia-iyer !
LGTM

**Pull request merged**